server/webapp/WEB-INF/rails.new/app/controllers/api_v1/users_controller.rb

```diff
@@ -43,12 +37,41 @@ def update
       end
     end
 
+    def create
+      @user_to_operate = user_service.findUserByName(params[:login_name])
+
+      unless @user_to_operate.instance_of?(com.thoughtworks.go.domain.NullUser)
```

Comment:
I din't mean to say use `addUserIfDoesNotExist`, but use `enableUserMutex` as used in this method to avoid race-conditions.